client.go

```diff
@@ -753,3 +761,15 @@ func (s *Client) Probe(ctx context.Context, req *ProbeRequest) error {
 		req.RouteHints,
 	)
 }
+
+// AbandonSwap sends the hash of the swap to abandon to all active swaps. Only
+// the addressed swap will abandon itself.
+func (s *Client) AbandonSwap(req *AbandonSwapRequest) error {
+	if req == nil {
+		return errors.New("no request provided")
+	}
+
+	s.abandonChan <- req.SwapHash
```

Comment:
I think you need to change the core structure here. A single channel won't notify all swaps that are waiting to read from it.

Here's an example:
We have 3 loop-ins active, `in1` `in2` and `in3`
All of them are waiting to read from `s.abandonChan` in case the user requests for one of them to be abandoned

User wants to abandon `in2` so they call `AbandonSwap` method and that method writes to `s.abandonChan` as shown in the line above.

A **random** swap will read that value **once**, and then the channel is going to be clear. Let's say that in the above example `in1` is reading the value, it checks that `in1` != `in2` and continues with a noop. The intended swap never received the value over the channel because it was consumed only once by `in1`.

Here's 2 ways you can proceed with this:
1. Enhance the "abandonChan" into a structure that is able to hold multiple channels for each subscriber (i.e you could have a `map[lntypes.Hash]chan Bool` which means one dedicated channel for each swap). When user tries to `AbandonSwap` we will first find the correct channel by accessing the map, and then try to write to that channel.

2. The more hacky approach is for each loop-in to re-write the value of `lntypes.Hash` into the channel when they check it and realise it doesn't match theirs. In that case the value would be written only once on the channel from this speciifc line above, but then it would ping-pong between all active loop-ins until it would **eventually** reach the intended target. (**hacky and dirty**, but only 1 line of code)

Reply:
Thank you for the thorough review and the nice catch. I prefer option 1.) to fix this.